pkg/verifier/cosign/trustpolicies.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosign
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+
+	re "github.com/deislabs/ratify/errors"
+)
+
+type TrustPolicies struct {
+	policies []TrustPolicy
+}
+
+const GlobalWildcardCharacter = '*'
+
+var validScopeRegex = regexp.MustCompile(`^[a-z0-9\.\-:@\/]*\*?$`)
+
+// CreateTrustPolicies creates a set of trust policies from the given configuration
+func CreateTrustPolicies(configs []TrustPolicyConfig, verifierName string) (*TrustPolicies, error) {
+	if configs == nil {
+		return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail("failed to create trust policies: no policies found")
+	}
+
+	if len(configs) == 0 {
+		return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail("failed to create trust policies: no policies defined")
+	}
+
+	policies := make([]TrustPolicy, 0, len(configs))
+	names := make(map[string]struct{})
+	for _, policyConfig := range configs {
+		if _, ok := names[policyConfig.Name]; ok {
+			return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail(fmt.Sprintf("failed to create trust policies: duplicate policy name %s", policyConfig.Name))
+		}
+		names[policyConfig.Name] = struct{}{}
+		policy, err := CreateTrustPolicy(policyConfig, verifierName)
+		if err != nil {
+			return nil, err
+		}
+		policies = append(policies, policy)
+	}
+
+	if len(policies) == 0 {
+		return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName).WithDetail("failed to create trust policies: no trust policies defined")
+	}
+
+	err := validateScopes(policies)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TrustPolicies{
+		policies: policies,
+	}, nil
+}
+
+// GetScopedPolicy returns the policy that applies to the given reference
+func (tps *TrustPolicies) GetScopedPolicy(reference string) (TrustPolicy, error) {
```

Comment:
yeah that's a good point. I did do this for the trust policy operation but not for trust policies. For Unit testing, I manually set the correct configs for trust policies so I didn't need mock. In future it could be helpful. I can leave it as a TODO?